storage/volume/interface.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package volume
+
+import "github.com/juju/names"
+
+// Disk
+type Disk interface {
+	// TODO(anastasiamac 2015-01-30 add name and persisted parameters
+	// when model is decided
+	Attachments() []Attachment
+}
+
+// Attachment is a block device
+type Attachment interface {
```

Comment:
FWIW, a lot of that information is specific to the volume, not the attachment. e.g. UUID, Label, Size, FilesystemType, Storage, Provisioned. I'm not too fussed, since this will be replaced soon by structures in the top-level storage package.